torch/distributions/transforms.py

```diff
@@ -600,6 +601,26 @@ def log_abs_det_jacobian(self, x, y):
         return -F.softplus(-x) - F.softplus(x)
 
 
+class SoftplusTransform(Transform):
+    r"""
+    Transform via the mapping :math:`\text{Softplus}(x) = \log(1 + \exp(x))`.
+    """
+    domain = constraints.real
+    codomain = constraints.positive
+
+    def __eq__(self, other):
+        return isinstance(other, SoftplusTransform)
+
+    def _call(self, x):
+        return x.exp().log1p()
```

Comment:
This is not stable for moderate `x` values. It is better to return `softplus(x)`

Reply:
@fehiepsi doesn't using pytorch `softplus` (with `beta=1, threshold=20`) change the derivative of the transform? E.g. when `threshold > 20` `log_abs_det_jacobian` should return 1.

Reply:
Do you mean log_abs_det return 0?

Using threshold looks fine to me though it seems to me that it is not needed. Probably, PyTorch doesn't use `logaddexp(0, x)` under the hood, so it needs the threshold...

Reply:
Yes, I mean that log_abs_det should return 0 when x > 20 because when x > 20 the transform is y=x not y=softplus(x). Maybe I am missing something.

Reply:
`-softplus(-20.)` returns `tensor(-2.0612e-09)`, which is good enough I guess.

Reply:
Makes sense to use approximation then.